src/main/java/carpettisaddition/helpers/rule/simpleNeighborUpdater/MutableNeighborUpdater.java

```diff
@@ -0,0 +1,58 @@
+package carpettisaddition.helpers.rule.simpleNeighborUpdater;
+
+import carpettisaddition.CarpetTISAdditionSettings;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+import net.minecraft.world.block.ChainRestrictedNeighborUpdater;
+import net.minecraft.world.block.NeighborUpdater;
+import net.minecraft.world.block.SimpleNeighborUpdater;
+import org.jetbrains.annotations.Nullable;
+
+public class MutableNeighborUpdater implements NeighborUpdater {
+    private final SimpleNeighborUpdater oldNeighborUpdater;
+    private final ChainRestrictedNeighborUpdater newNeighborUpdater;
+
+    public MutableNeighborUpdater(World world, ChainRestrictedNeighborUpdater updater) {
+        this.oldNeighborUpdater = new SimpleNeighborUpdater(world);
+        this.newNeighborUpdater = updater;
+    }
+
+    @Override
+    public void replaceWithStateForNeighborUpdate(Direction direction, BlockState neighborState, BlockPos pos, BlockPos neighborPos, int flags, int maxUpdateDepth) {
```

Comment:
Are you sure that this implementation is stable when the rule value is changed during the update chain?

Reply:
My fault, I considered it but I could not find a good solution.

Hot switching is always a big problem (Like sand spawner). I reviewed codes of Carpet-Fixes, it only constructs a new updater when hot switching. The only difference between us is there may be some updates in my stack or queue when a new update comes.

The only way to deal with hot switching is preventing hot switching. It means switch can happen only when stack or queue is empty. It is easy when switching from new updater to old one by detecting `depth` in `ChainRestrictedNeighborUpdater`, but it is not easy to reverse, how to detect the size of invoking stack? I think setting a global counter and resetting it when catching exception is workable.